react/features/background/middleware.js

```diff
@@ -0,0 +1,85 @@
+/* @flow */
+
+import { AppState } from 'react-native';
+import type { Dispatch } from 'redux';
+
+import { appStateChanged, setBackgroundVideoMuted } from './actions';
+import { APP_STATE_CHANGED } from './actionTypes';
+
+import {
+    APP_WILL_MOUNT,
+    APP_WILL_UNMOUNT
+} from '../app';
+import { MiddlewareRegistry } from '../base/redux';
+
+
+/**
+ * Holds the callback function we need for the AppState event listener, which
+ * is a partial of __onAppStateChanged with the added dispatch function
+ * parameter.
+ */
+let _onAppStateChanged;
```

Comment:
I do not want globals. I realize that `AppState` is a global but the `store.dispatch` in `_onAppStateChanged = __onAppStateChanged.bind(undefined, store.dispatch)` is not a global. So what's happening is that you're assigning a non-global to a global.